src/gateway/tasks/syncBlocks.ts

```diff
@@ -63,6 +63,9 @@ interface ReqVariables {
 
 
 export async function runSyncBlocksTask(context: Application.BaseContext) {
+  // Hmm, not sure if this approach is much better, as it can result in fastly increasing memory usage
```

Comment:
It's not a priority and we probably shouldn't implement it now. What do you guys think about abstracting the execution in a job queue, like https://github.com/bee-queue/bee-queue

1) Pros -> better control over concurrency, memory, debugging, (most importantly being able to gracefully continue after a shutdown)
2) Cons -> added complexity, too much enterprise patterns, additional components (redis)

Reply:
re. greacefully continuing after a shutdown - currently db gives us such feature..we're simply searching for the last processed block height at (after starting the gateway again) continue the processing..same in case of verifying interactions...
One place where it could be useful that I can think of is:
One process puts new blocks/interactions into queue (queueueueue https://www.youtube.com/watch?v=0V9Ua538jMI&t=223s ;-)) and there are multiple process/nodes that take values from the queue in order to verify them. Currently it's a bit hard to introduce such parallelism, because each node would start to verify the transactions at (roughly) the exact same point (so they would basically overlap their work) - I had to introduce some manual block height offsets in the sql queries when I was in the process of verifying all the interactions by a set of 12 nodes.

Reply:
let's move the conversation to https://github.com/redstone-finance/redstone-sw-gateway/issues/16